server/punishments.ts

```diff
@@ -861,17 +864,22 @@ export const Punishments = new class {
 		}
 		return success;
 	}
-	automaticUnlock(user: User, punishedID: ID | PunishType) {
-		if (user.locked === punishedID) {
-			Punishments.unlock(user.id);
-		}
-	}
 	/**
-	 * Validates the length and sets the punishment timer for a user.
+	 * Sets the punishment timer for a user,
+	 * to either MAX_PUNISHMENT_TIMER_LENGTH or the amount of time left on the punishment.
+	 * It also expires a punishment if the time is up.
 	 */
-	setPunishmentTimer(user: User, expireTime: number, punishedID: ID | PunishType) {
-		if (expireTime === 0 || expireTime === Infinity || expireTime > Chat.MAX_TIMEOUT_DURATION) return;
-		user.punishmentTimer = setTimeout(Punishments.automaticUnlock, expireTime - Date.now(), user, punishedID);
+	checkPunishmentTime(user: User, punishment: Punishment) {
+		const [, id, expireTime] = punishment;
+
+		if (expireTime < Date.now()) {
```

Comment:
```suggestion
		if (expireTime - 1 <= Date.now()) {
```

A bit more leeway so you don't set timers for 1ms if there's some minor timer jitter.